### ISSUE TITLE
no-std support for Rust

### ISSUE THREAD

**Issue by stevefan1999-personal** [cenotelie/hime#83]
I want to use this in a compiler project but the core part of the compiler, namely the parsing part is split out as `no_std` code and only specific loading algorithms are std only. This makes it quite difficult for Rust Embedding community as well

**Comment by woutersl** [comment]
Hi, `no_std` could be a reachable goal if `alloc` is still amenable to you. I think removing all allocation is impossible, LR and GLR algorithms were not designed for this use case.

**Comment by stevefan1999-personal** [comment]
@woutersl Yes that would be wonderful, given that we just need to have a static buffer to declare as a heap memory blob, having alloc is fine.

**Comment by woutersl** [comment]
@stevefan1999-personal , I just published a version 4.3.0 of [hime_redist](https://crates.io/crates/hime_redist), with support for `no_std`. There is a "Support for no_std" paragraph in the package description showing how to setup the dependency for this purpose (which is just the classic having `std` feature that can be deactivated). `alloc` is still required as previously mentioned.

**Comment by stevefan1999-personal** [comment]
@woutersl Much appreciated!

**Comment by stevefan1999-personal** [comment]
@woutersl I also want to see if the codegen part could support no_std as well, in particular it seems like `use std::io::Read` is emitted every time and need some manual intervention. I intent to use the SDK to directly generate the code every time you compile the parser, but std::io is not emigrated to core::io yet in the current time, so it is quite hard to decide.
